6_visualize/src/prep_spark_line_fun.R

```diff
@@ -0,0 +1,71 @@
+
+    y_pos <- y_start # initialize position of first spark
+    for(i in sites) {
+
+      # Filter data to just one site & create polygon out of it
+      storm_data_i <- filter(this_spark, site_no == i) %>% arrange(dateTime)
+      storm_data_site_all <- filter(storm_data, site_no == i)
+
+      if(nrow(storm_data_i) == 0) { next } # skip if there's no data at or before this timestep
+
+      flood_stage_va <- as.numeric(unique(storm_data_i$flood_stage_normalized))
+      lowest_stage <- min(storm_data_site_all$stage_normalized, na.rm = TRUE)
+      highest_stage <- max(storm_data_site_all$stage_normalized, na.rm = TRUE)
+
+      full_polygon <- data.frame(dateTime = head(storm_data_i$dateTime, 1),
+                                 stage_normalized = min(storm_data_i$stage_normalized,1)) %>%
+        bind_rows(select(storm_data_i, dateTime, stage_normalized)) %>%
+        bind_rows(data.frame(dateTime = tail(storm_data_i$dateTime, 1),
+                             stage_normalized = min(storm_data_i$stage_normalized)))
+
+      # Fill values lower than flood stage with the stage & then create a polygon out of it
+      above_flood_data <- mutate(storm_data_i, stage_normalized = ifelse(stage_normalized < flood_stage_va,
+                                                                         flood_stage_va,
+                                                                         stage_normalized))
+      above_flood_polygon <- data.frame(dateTime = head(above_flood_data$dateTime, 1),
+                                        stage_normalized = head(above_flood_data$stage_normalized, 1)) %>%
+        bind_rows(select(above_flood_data, dateTime, stage_normalized)) %>%
+        bind_rows(data.frame(dateTime = tail(above_flood_data$dateTime, 1),
+                             stage_normalized = min(above_flood_data$stage_normalized)))
+
+      y_coords <- c(y_pos, y_pos + vertical_spacing) # y coords relative to plot box
+      fig.new <- c(grconvertX(x_coords, from="npc", to="ndc"), grconvertY(y_coords, from="npc", to="ndc"))
+
+      op <- par(fig=fig.new, cex=2, new=TRUE, mar=rep(0, 4)) # setup parameters
+      # setup a plotting device
+      plot(x=NA, y=NA, type='n', axes=FALSE, xlab="", ylab="",
+           xlim = c(as.POSIXct(viz_config$dates$start, tz = "UTC"), as.POSIXct(viz_config$dates$end, tz = "UTC")),
+           ylim = c(lowest_stage, highest_stage))
+      # put stage polygons on plot
+      polygon(full_polygon$dateTime, full_polygon$stage_normalized, col = viz_config$gage_norm_col, border=NA)
+      polygon(above_flood_polygon$dateTime, above_flood_polygon$stage_normalized, col = viz_config$gage_flood_col, border=NA)
+      points(full_polygon$dateTime[-nrow(full_polygon)], full_polygon$stage_normalized[-nrow(full_polygon)],
+             col = viz_config$gage_flood_col, type='l', lwd=2.5)
+      par(op) # reset plot parameters
+
+      y_pos <- y_pos + vertical_spacing # increment spacing for next plot
```

Comment:
Which order are sites getting plotted in? From top to bottom or bottom to top? I forget whether y coordinates in R ever start at the top. 

I suspect it'd be easiest to interpret & modify this if the order were from top to bottom, because I think if and when we want to order the sparklines by some variable, our heads will be in dplyr mode and top to bottom will make most sense.

Reply:
I did not try to do any particular order at the moment. Not really sure where that belongs? Here I suppose.